task10/part2/src/Item.ts

```diff
@@ -0,0 +1,28 @@
+import { Pages } from "./pages";
+
+export abstract class Item {
+    pages: Pages;
+
+    constructor(pages: Pages) {
+        this.pages = pages;
+    }
+
+    abstract toString(): string;
+
+    [Symbol.iterator]() {  
+        let index = 0;
+
+        return {
+            next: () => {
+                if (index < this.pages.length) {
+                    let outputString = this.toString();
+                    outputString += `, ${this.pages.printPageDetail(index)}`;
+                    index++;
+                    return { value: outputString, done: false }
+                } else {
+                    return { value: '', done: true }
+                }
+            }
+        }
+    }
```

Comment:
I would proposed put this logic into PagesIterablesMixins
And also it could be simplified
Please, see the way to create an iterator here
https://stackoverflow.com/questions/52003806/creating-class-as-iterable-in-angular-typescript

After that you could simply extend Item class with PagesIterablesMixins